### PR TITLE
dns/dyndns: add support for 1984Hosting

### PR DESCRIPTION
**Pull request by phedoreanu**
This PR adds DynDNS support for [1984Hosting](https://1984hosting.com/product/freedns/).

### CODE REVIEW

**Comment by Starkstromkonsument** [comment]
Hi, please take a look at https://github.com/opnsense/plugins/issues/1611#issuecomment-706129596 and #2053. As you are also interested in contributing to the dyndns-plugin, would you like to help? Please join the discussion at #1611 if you are interested. Thank you.

File: dns/dyndns/src/etc/inc/plugins.inc.d/dyndns.inc [hunk 1]
```diff
@@ -96,6 +96,7 @@ function dyndns_list()
      */
 
     return array(
+        '1984Hosting' => '1984Hosting',
```
Comment (fichtner) [inline dns/dyndns/src/etc/inc/plugins.inc.d/dyndns.inc:99]:
```suggestion
        '1984-hosting' => '1984 Hosting Company',
```

File: dns/dyndns/src/etc/inc/plugins.inc.d/dyndns/phpDynDNS.inc [hunk 2]
```diff
@@ -14,6 +14,7 @@
  *    - _debug()
  *    - _checkIP()
  * +----------------------------------------------------+
+ *  1984Hosting                 - Last Tested: 03 August 2020
```
Comment (fichtner) [inline dns/dyndns/src/etc/inc/plugins.inc.d/dyndns/phpDynDNS.inc:17]:
```suggestion
 *  1984 Hosting Company         - Last Tested: 03 August 2020
```

File: dns/dyndns/src/etc/inc/plugins.inc.d/dyndns/phpDynDNS.inc [hunk 3]
```diff
@@ -168,6 +169,7 @@ class updatedns
             case 'linode':
             case 'linode-v6':
             case 'namecheap':
+            case '1984Hosting':
```
Comment (fichtner) [inline dns/dyndns/src/etc/inc/plugins.inc.d/dyndns/phpDynDNS.inc:172]:
```suggestion
            case '1984-hosting':
```

File: dns/dyndns/src/etc/inc/plugins.inc.d/dyndns/phpDynDNS.inc [hunk 4]
```diff
@@ -304,6 +306,7 @@ class updatedns
             $this->_error(10);
         } else {
             switch ($this->_dnsService) {
+                case '1984Hosting':
```
Comment (fichtner) [inline dns/dyndns/src/etc/inc/plugins.inc.d/dyndns/phpDynDNS.inc:309]:
```suggestion
                case '1984-hosting':
```

File: dns/dyndns/src/etc/inc/plugins.inc.d/dyndns/phpDynDNS.inc [hunk 5]
```diff
@@ -1031,6 +1034,11 @@ class updatedns
                     curl_setopt($ch, CURLOPT_IPRESOLVE, CURL_IPRESOLVE_V4);
                 }
                 break;
+            case '1984Hosting':
```
Comment (fichtner) [inline dns/dyndns/src/etc/inc/plugins.inc.d/dyndns/phpDynDNS.inc:1037]:
```suggestion
            case '1984-hosting':
```

File: dns/dyndns/src/etc/inc/plugins.inc.d/dyndns/phpDynDNS.inc [hunk 6]
```diff
@@ -1643,6 +1651,20 @@ class updatedns
                         $this->_debug($data);
                 }
                 break;
+            case '1984Hosting':
```
Comment (fichtner) [inline dns/dyndns/src/etc/inc/plugins.inc.d/dyndns/phpDynDNS.inc:1654]:
```suggestion
            case '1984-hosting':
```

File: dns/dyndns/src/etc/inc/plugins.inc.d/dyndns/phpDynDNS.inc [hunk 7]
```diff
@@ -1643,6 +1651,20 @@ class updatedns
                         $this->_debug($data);
                 }
                 break;
+            case '1984-hosting':
+                $output = json_decode($data);
+                if ($output->ok === true) {
+                    $status = "Dynamic DNS: (Success) {$this->_dnsHost} updated to {$this->_dnsIP}";
+                    $successful_update = true;
+                } elseif ($output->error) {
+                    $status = "Dynamic DNS ({$this->_dnsHost}): ERROR - Invalid Credentials! Don't forget to use API Key for password field with 1984Hosting.";
```
Comment (fichtner) [inline dns/dyndns/src/etc/inc/plugins.inc.d/dyndns/phpDynDNS.inc:1660]:
```suggestion
                    $status = "Dynamic DNS ({$this->_dnsHost}): ERROR - Invalid Credentials! Don't forget to use API Key for password field with 1984 Hosting Company.";
```

File: dns/dyndns/src/www/services_dyndns_edit.php [hunk 8]
```diff
@@ -128,6 +129,7 @@ function is_dyndns_username($uname)
             case 'linode':
             case 'linode-v6':
             case 'namecheap':
+            case '1984Hosting':
```
Comment (fichtner) [inline dns/dyndns/src/www/services_dyndns_edit.php:132]:
```suggestion
            case '1984-hosting':
```

File: dns/dyndns/src/www/services_dyndns_edit.php [hunk 9]
```diff
@@ -75,7 +75,8 @@ function is_dyndns_username($uname)
     }
     $input_errors = array();
     $pconfig = $_POST;
-    if(($pconfig['type'] == "freedns" || $pconfig['type'] == "linode" || $pconfig['type'] == "linode-v6" || $pconfig['type'] == "namecheap" || $pconfig['type'] == "cloudflare-token" || $pconfig['type'] == "cloudflare-token-v6" || $pconfig['type'] == "desec" || $pconfig['type'] == "desec-v4-v6" || $pconfig['type'] == "desec-v6") && $pconfig['username'] == "") {
+
+    if(($pconfig['type'] == "freedns" || $pconfig['type'] == "linode" || $pconfig['type'] == "linode-v6" || $pconfig['type'] == "namecheap" || $pconfig['type'] == "cloudflare-token" || $pconfig['type'] == "cloudflare-token-v6" || $pconfig['type'] == "desec" || $pconfig['type'] == "desec-v4-v6" || $pconfig['type'] == "desec-v6" || $pconfig['type'] == "1984Hosting") && $pconfig['username'] == "") {
```
Comment (fichtner) [inline dns/dyndns/src/www/services_dyndns_edit.php:79]:
```suggestion
    if(($pconfig['type'] == "freedns" || $pconfig['type'] == "linode" || $pconfig['type'] == "linode-v6" || $pconfig['type'] == "namecheap" || $pconfig['type'] == "cloudflare-token" || $pconfig['type'] == "cloudflare-token-v6" || $pconfig['type'] == "desec" || $pconfig['type'] == "desec-v4-v6" || $pconfig['type'] == "desec-v6" || $pconfig['type'] == "1984-hosting") && $pconfig['username'] == "") {
```

File: dns/dyndns/src/etc/inc/plugins.inc.d/dyndns/phpDynDNS.inc [hunk 10]
```diff
@@ -1643,6 +1651,20 @@ class updatedns
                         $this->_debug($data);
                 }
                 break;
+            case '1984-hosting':
+                $output = json_decode($data);
+                if ($output->ok === true) {
```
Comment (fichtner) [inline dns/dyndns/src/etc/inc/plugins.inc.d/dyndns/phpDynDNS.inc:1656]:
```suggestion
                if ($output === null) {
                    /* empty response */
                } elseif ($output->ok === true) {
```

File: dns/dyndns/src/etc/inc/plugins.inc.d/dyndns/phpDynDNS.inc [hunk 11]
```diff
@@ -1643,6 +1651,22 @@ class updatedns
                         $this->_debug($data);
                 }
                 break;
+            case '1984-hosting':
+                $output = json_decode($data);
+                if ($output === null) {
+                    /* empty response */
```
Comment (fichtner) [inline dns/dyndns/src/etc/inc/plugins.inc.d/dyndns/phpDynDNS.inc:1657]:
```suggestion
                    $status = "Dynamic DNS ({$this->_dnsHost}): ERROR - empty response.";
```

**Comment by fichtner** [comment]
Merged, thanks!

**Pull request merged**